### PR TITLE
Upgrade greenplum dialect

### PR DESCRIPTION
**Pull request by kkozhakin**
### Brief summary of the change made

- add combine version of interval for ansi dialect;
- add greenplum keywords;
- add fetch, analize, declare, close, copy statements for greenplum dialect;
- fix create statement in greenplum dialect;
- add walrus operator, drop aggregate statement for postgres dialect;
- fix FunctionContentsGrammar, DateTimeLiteralGrammar, overlaps segment in postgres dialect;

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by kkozhakin** [comment]
Maybe, copy statement will work from https://github.com/sqlfluff/sqlfluff/pull/5181

File: src/sqlfluff/dialects/dialect_ansi.py
Comment (kkozhakin) [inline src/sqlfluff/dialects/dialect_ansi.py:1]: Maybe there is a better way to fix this bug.

Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_ansi.py:1]: Have you got something in mind?

Comment (kkozhakin) [reply src/sqlfluff/dialects/dialect_ansi.py:1]: Nothing yet(

**Review by WittierDinosaur — APPROVED** [review]
In the future, it'd be super useful for the reviewers if you could split things out so that it's one PR per feature. But this all looks good, thanks!! LGTM

**Pull request merged**